examples/vision/consistency_training.py

```diff
@@ -0,0 +1,391 @@
+    .map(lambda x, y: (preprocess_train(x, y, noisy=False)), num_parallel_calls=AUTO)
+    .batch(BATCH_SIZE)
+    .prefetch(AUTO)
+)
+
+# This prepares the `Dataset` object to use RandAugment.
+train_noisy_ds = (
+    train_ds.shuffle(BATCH_SIZE * 10, seed=42)
+    .map(preprocess_train, num_parallel_calls=AUTO)
+    .batch(BATCH_SIZE)
+    .prefetch(AUTO)
+)
+
+validation_ds = (
+    validation_ds.map(preprocess_test, num_parallel_calls=AUTO)
+    .batch(BATCH_SIZE)
+    .prefetch(AUTO)
+)
+
+test_ds = (
+    test_ds.map(preprocess_test, num_parallel_calls=AUTO)
+    .batch(BATCH_SIZE)
+    .prefetch(AUTO)
+)
+
+# This dataset will be used to train the second model.
+consistency_training_ds = tf.data.Dataset.zip((train_clean_ds, train_noisy_ds))
+
+"""
+## Visualize the datasets
+"""
+
+sample_images, sample_labels = next(iter(train_clean_ds))
+plt.figure(figsize=(10, 10))
+for i, image in enumerate(sample_images[:9]):
+    ax = plt.subplot(3, 3, i + 1)
+    plt.imshow(image.numpy().astype("int"))
+    plt.axis("off")
+
+sample_images, sample_labels = next(iter(train_noisy_ds))
+plt.figure(figsize=(10, 10))
+for i, image in enumerate(sample_images[:9]):
+    ax = plt.subplot(3, 3, i + 1)
+    plt.imshow(image.numpy().astype("int"))
+    plt.axis("off")
+
+"""
+## Define a model building utility function
+
+We now define our model building utility. Our model is based on the [ResNet50V2 architecture](https://arxiv.org/abs/1603.05027).
+"""
+
+
+def get_training_model(num_classes=10):
+    resnet50_v2 = tf.keras.applications.ResNet50V2(
+        weights=None,
+        include_top=False,
+        input_shape=(CROP_TO, CROP_TO, 3),
+    )
+    model = tf.keras.Sequential(
+        [
+            layers.Input((CROP_TO, CROP_TO, 3)),
+            layers.experimental.preprocessing.Rescaling(scale=1.0 / 127.5, offset=-1),
+            resnet50_v2,
+            layers.GlobalAveragePooling2D(),
+            layers.Dense(num_classes),
+        ]
+    )
+    return model
+
+
+"""
+In the interest of reproducibility, we serialize the initial random weights of the
+teacher  network.
+"""
+
+initial_teacher_model = get_training_model()
+initial_teacher_model.save_weights("initial_teacher_model.h5")
+
+"""
+## Train the teacher model
+
+As noted in Noisy Student Training, if the teacher model is trained with *geometric
+ensembling* and when the student model is forced to mimic that, it leads to better
+performance. The original work uses [Stochastic Depth](https://arxiv.org/abs/1603.09382)
+and [Dropout](https://jmlr.org/papers/v15/srivastava14a.html) to bring in the ensembling
+part but for this example, we will use [Stochastic Weight Averaging](https://arxiv.org/abs/1803.05407)
+(SWA) which also resembles geometric ensembling.
+"""
+
+# Define the callbacks.
+reduce_lr = tf.keras.callbacks.ReduceLROnPlateau(patience=3)
+es = tf.keras.callbacks.EarlyStopping(patience=10, restore_best_weights=True)
+
+# Initialize SWA from tf-hub.
+SWA = tfa.optimizers.SWA
```

Comment:
How are you using this?

Reply:
Yes. I had forgotten to use it to wrap the optimizer. It's been fixed now and I have also added a comment above its usage so that it won't go unnoticed.